doc/console/serial-console-HLD.md

```diff
@@ -0,0 +1,230 @@
+# Serial-console global config HLD
+##  Table of Content 
+- [Serial-console config HLD](#serial-console-global-config-hld)
+- [Table of Content](#table-of-content)
+- [Revision](#revision)
+- [1. Introduction](#1-introduction)
+- [1.1. Scope](#11-scope)
+- [1.2. Definitions/Abbreviations](#12-definitionsabbreviations)
+- [1.3. Overview](#13-overview)
+- [1.4. Requirements](#14-requirements)
+- [2. Architecture Design](#2-architecture-design)
+    - [2.1. Configuration modules](#21-configuration-modules)
+- [3. High-Level Design](#3-high-level-design)
+      - [Flow diagram](#flow-diagram)
+    - [3.1 Flow description](#31-flow-description)
+    - [3.2 serial-console configurations](#32-serial-console-configurations)
+- [4. Init flow](#4-init-flow)
+    - [4.1. Feature Default](#41-feature-default)
+- [5. SAI api](#5-sai-api)
+- [6. Configuration and management](#6-configuration-and-management)
+    - [6.1. SERIAL_CONSOLE configDB table](#61-serial_console-configdb-table)
+    - [6.2. ConfigDB schemas](#62-configdb-schemas)
+    - [6.3. CLI/YANG model Enhancements](#63-cliyang-model-enhancements)
+    - [6.4. Config DB Enhancements](#64-config-db-enhancements)
+    - [6.5. Manifest (if the feature is an Application Extension)](#65-manifest-if-the-feature-is-an-application-extension)
+- [7. Warmboot and Fastboot Design Impact](#7-warmboot-and-fastboot-design-impact)
+- [8. Restrictions/Limitations](#8-restrictionslimitations)
+- [9. Testing Requirements/Design](#9-testing-requirementsdesign)
+    - [9.1. Unit Test cases](#91-unit-test-cases)
+    - [9.2. System Test cases](#92-system-test-cases)
+- [10. Open/Action items - if any](#10-openaction-items---if-any)
+###  Revision  
+
+|  Rev  |   Date   |   Author   | Change Description |
+| :---: | :------: | :--------: | ------------------ |
+|  0.1  | 16/06/23 | Ivan Davydenko | Initial version    |
+
+##  1. Introduction  
+
+###  1.1. Scope  
+
+This hld doc for serial cli global configurations describes the requirements, architecture and general flow details of serial connection config in SONIC OS based switches.
+
+###  1.2. Definitions/Abbreviations 
+
+	serial - secure shell
```

Comment:
>secure shell [](http://example.com/codeflow?start=10&length=12)

this is for ssh?

Reply:
No, this section covers the serial connection. 
Fixed in 9e32292.